### PR TITLE
make use of ledger-geoid and the /v1/exchange/providers endpoint

### PR DESCRIPTION
**Pull request by mrose17**
auditor: @diracdeltas 

test plan: nothing to be done until https://github.com/brave/browser-laptop/issues/3711 resolved

fixes https://github.com/brave/browser-laptop/pull/3824

### CODE REVIEW

File: docs/state.md [hunk 1]
```diff
@@ -435,8 +435,14 @@ WindowStore
       }
     },
     hasBitcoinHandler: boolean,  // brave browser has a `bitcoin:` URI handler
-    paymentIMG: string,           // the QR code equivalent of `paymentURL` expressed as "data:image/...;base64,..."
-    error: { // non-null if the last updateLedgerInfo happened concurrently with an error
+    countryCode: string,         // ISO3166 2-letter code for country of browser's location
+    exchangeInfo: {              // information about corresponding "friendliest" BTC exchange (suggestions welcome!)
+      titleText: string,         // e.g., 'Create an account on ...'
+      buttonText: string,        // e.g., 'Create'
```
Comment (diracdeltas) [inline docs/state.md]: as mentioned in review, please replace button and titletext with just exchangeName

Comment (mrose17) [inline docs/state.md]: as mentioned in my reply: i don't think that's a good idea, but let's give it a shot and see what happens.

File: app/ledger.js [hunk 2]
```diff
@@ -784,13 +784,10 @@ var updateLedgerInfo = () => {
...
       roundtrip({ path: '/v1/exchange/providers' }, client.options, (err, response, body) => {
-        var exchangeInfo
-
         if (err) console.log('ledger exchange error: ' + JSON.stringify(err, null, 2))
 
-        exchangeInfo = (body || {})[ledgerInfo.countryCode]
-        if (exchangeInfo) ledgerInfo.exchangeURL = exchangeInfo.buttonURL
-        updateLedgerInfo()
+        ledgerInfo._internal.exchanges = body || {}
+        ledgerInfo.exchangeInfo = ledgerInfo._internal.exchanges[ledgerInfo.countryCode]
```
Comment (diracdeltas) [inline app/ledger.js]: does this need a call to `updateLedgerInfo`?

**Comment by diracdeltas** [comment]
thanks

**Comment by luixxiul** [comment]
Is QA needed?

**Comment by mrose17** [comment]
thanks for checking. no qa needed!

**Pull request merged**